src/main/scala/beam/utils/ProfilingUtils.scala

```diff
@@ -15,4 +15,10 @@ object ProfilingUtils {
     val endTime = System.currentTimeMillis()
     endTime - startTime
   }
+
+  def timed[U](what: String, work: => U, logger: String => Unit): U = {
```

Comment:
What about changing the method signature to:

    def timed[U](what: String, logger: String => Unit)(work: () => U): U = {

As you can then use it like this:

    time("what", x=>log.info(x)){ () => 
      //inline method
    }

Not a huge difference, so take it or leave it. I just know that this is often the pattern for this type of method.